SignalServiceKit/src/Protos/Generated/SSKProto.swift

```diff
@@ -0,0 +1,3588 @@
+    }
+
+    private class func SSKProtoEnvelopeTypeWrap(_ value: SignalServiceProtos_Envelope.TypeEnum) -> SSKProtoEnvelopeType {
+        switch value {
+        case .unknown: return .unknown
+        case .ciphertext: return .ciphertext
+        case .keyExchange: return .keyExchange
+        case .prekeyBundle: return .prekeyBundle
+        case .receipt: return .receipt
+        }
+    }
+
+    private class func SSKProtoEnvelopeTypeUnwrap(_ value: SSKProtoEnvelopeType) -> SignalServiceProtos_Envelope.TypeEnum {
+        switch value {
+        case .unknown: return .unknown
+        case .ciphertext: return .ciphertext
+        case .keyExchange: return .keyExchange
+        case .prekeyBundle: return .prekeyBundle
+        case .receipt: return .receipt
+        }
+    }
+
+    // MARK: - SSKProtoEnvelopeBuilder
+
+    @objc public class SSKProtoEnvelopeBuilder: NSObject {
+
+        private var proto = SignalServiceProtos_Envelope()
+
+        @objc public override init() {}
+
+        @objc public func setType(_ valueParam: SSKProtoEnvelopeType) {
+            proto.type = SSKProtoEnvelopeTypeUnwrap(valueParam)
+        }
+
+        @objc public func setSource(_ valueParam: String) {
+            proto.source = valueParam
+        }
+
+        @objc public func setSourceDevice(_ valueParam: UInt32) {
+            proto.sourceDevice = valueParam
+        }
+
+        @objc public func setRelay(_ valueParam: String) {
+            proto.relay = valueParam
+        }
+
+        @objc public func setTimestamp(_ valueParam: UInt64) {
+            proto.timestamp = valueParam
+        }
+
+        @objc public func setLegacyMessage(_ valueParam: Data) {
+            proto.legacyMessage = valueParam
+        }
+
+        @objc public func setContent(_ valueParam: Data) {
+            proto.content = valueParam
+        }
+
+        @objc public func build() throws -> SSKProtoEnvelope {
+            let wrapper = try SSKProtoEnvelope.parseProto(proto)
+            return wrapper
+        }
+    }
+
+    fileprivate let proto: SignalServiceProtos_Envelope
+
+    @objc public let type: SSKProtoEnvelopeType
+    @objc public let source: String
+    @objc public let sourceDevice: UInt32
+    @objc public let timestamp: UInt64
+
+    @objc public var relay: String? {
+        guard proto.hasRelay else {
+            return nil
+        }
+        return proto.relay
+    }
+    @objc public var hasRelay: Bool {
+        return proto.hasRelay
+    }
+
+    @objc public var legacyMessage: Data? {
+        guard proto.hasLegacyMessage else {
+            return nil
+        }
+        return proto.legacyMessage
+    }
+    @objc public var hasLegacyMessage: Bool {
+        return proto.hasLegacyMessage
+    }
+
+    @objc public var content: Data? {
+        guard proto.hasContent else {
+            return nil
+        }
+        return proto.content
+    }
+    @objc public var hasContent: Bool {
+        return proto.hasContent
+    }
```

Comment:
That is: for strings, we can just `guard let someString = wrapper.someString else {` but for primitive types like `UInt64`, we _have to_ consult `has...` since the accessors will return a default value if the field isn't set. This is how our old obj-c and new Swift proto parsing code works, and I think it's better than trying to use `nullable NSNumber *` which loses the numeric type info.

Reply:
I understand why that would be true when the caller is in objc, but I'm not sure why it would be true for swift, where all types can be optional.

Reply:
These wrapper classes need to present a single API to Obj-c and Swift code.  I suppose we could offer a Swift-only accessor that is optional.  Almost all of our code that uses these wrappers is Obj-C, so I haven't bothered but we could add that.